src/scene_graph/scene_graph.cpp

```diff
@@ -90,6 +90,36 @@ namespace wr
 		}
 	}
 
+	void SceneGraph::UpdateSkyboxNode(TextureHandle new_equirectangular)
+	{
+		auto skybox_node = GetCurrentSkybox();
+
+		skybox_node->m_irradiance.value().m_pool->MarkForUnload(skybox_node->m_irradiance.value(), m_render_system->GetFrameIdx());
+		skybox_node->m_irradiance = std::nullopt;
+
+		skybox_node->m_skybox.value().m_pool->MarkForUnload(skybox_node->m_skybox.value(), m_render_system->GetFrameIdx());
+		skybox_node->m_skybox = std::nullopt;
+
+		skybox_node->m_prefiltered_env_map.value().m_pool->MarkForUnload(skybox_node->m_prefiltered_env_map.value(), m_render_system->GetFrameIdx());
+		skybox_node->m_prefiltered_env_map = std::nullopt;
+
+
+		if (skybox_node->m_hdr.m_pool)
+		{
+			skybox_node->m_hdr.m_pool->MarkForUnload(skybox_node->m_hdr, m_render_system->GetFrameIdx());
+
+#ifdef _DEBUG
+			//Decide if we want to break when developing in case we enter this function
+			//as in theory m_hdr needed to be cleaned from the render task, if that's not the case something went wrong.
+			LOGC("SOFT-ERROR: M_HDR is supposed to be an invalid handle at this stage. If that's not the case, the next line of code could leak memory");
```

Comment:
Not sure how to call it....the code doesn't break if the handle is not invalid, but I kinda want to be able to see that in the console during development because it means that someone is doing something strange with the code